test/tests/functional/pbs_provisioning_enhancement.py

```diff
@@ -0,0 +1,310 @@
+        j.set_attributes({'Resource_List.select': '1:aoe=osimage1'})
+        jid = self.server.submit(j)
+
+        # Job will start and wait for provisioning to complete.
+        self.server.expect(JOB, {ATTR_substate: '71'}, id=jid)
+        self.server.log_match("fake os provisioning script",
+                              max_attempts=60,
+                              interval=1)
+        self.momA.restart()
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+    def test_subchunk_application_provisioning(self):
+        """
+        Test application provisioning job request consist of subchunks
+        with and without aoe resource.
+        """
+
+        j = Job(TEST_USER1)
+        j.set_attributes({'Resource_List.select':
+                          '1:ncpus=1:aoe=App1+1:ncpus=12'})
+        j.set_sleep_time(5)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+        self.server.expect(JOB, ATTR_execvnode, id=jid, op=SET)
+        nodes = j.get_vnodes(j.exec_vnode)
+        self.server.log_match("fake application provisioning script",
+                              max_attempts=20,
+                              interval=1)
+        self.assertTrue((nodes[0] == self.momA.shortname and
+                         nodes[1] == self.momB.shortname) or
+                        (nodes[0] == self.momB.shortname and
+                         nodes[1] == self.momA.shortname))
+
+    def test_subchunk_os_provisioning(self):
+        """
+        Test os provisioning job request consist of subchunks
+        with and without aoe resource.
+        """
+
+        a = {'Resource_List.select': '1:aoe=osimage1+1:ncpus=12'}
+        j = Job(TEST_USER1, a)
+        j.set_sleep_time(10)
+        jid = self.server.submit(j)
+        self.server.expect(JOB, ATTR_execvnode, id=jid, op=SET)
+        nodes = j.get_vnodes(j.exec_vnode)
+        self.assertTrue((nodes[0] == self.momA.shortname and
+                         nodes[1] == self.momB.shortname) or
+                        (nodes[0] == self.momB.shortname and
+                         nodes[1] == self.momA.shortname))
+        self.momA.restart()
+        self.server.expect(JOB, {ATTR_state: 'R'}, id=jid)
+
+    def test_job_wide_provisioining_request(self):
+        """
+        Test jobs with jobwide aoe resource request.
+        """
+
+        # Below job will not run, since resource requested are job-wide,
+        # and no single node have all the requested resource.
+
+        j = Job(TEST_USER1)
+        j.set_sleep_time(5)
+        j.set_attributes({"Resource_List.aoe": "App1",
+                          "Resource_List.ncpus": 12})
+        jid = self.server.submit(j)
+        self.server.expect(JOB, {ATTR_state: 'Q',
+                                 ATTR_comment:
+                                 (MATCH_RE, 'Not Running: Insufficient ' +
+                                  'amount of resource*')}, id=jid)
```

Comment:
The * at the end won't do what you want.  It will match zero or more e's.  You want .*